### PR TITLE
feat: BI-5954 BaseEntrySchemaMigration class

### PR DESCRIPTION
**Pull request by ForrestGump**

### CODE REVIEW

File: lib/dl_core/dl_core/us_manager/schema_migration/base.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+@attr.s
+class BaseEntrySchemaMigration:
+    _MIGRATIONS: ClassVar[list[Migration]] = attr.ib(init=False, factory=list)
+    strict_migration: bool = attr.ib(default=False)
+
+    @classmethod
+    def _collect_migrations(cls) -> list[Migration]:
+        migrations: list[Migration] = []
+        for _cls in cls.__mro__:
```
Comment (ovsds) [inline lib/dl_core/dl_core/us_manager/schema_migration/base.py:39]: ooof, mb better use `super().` mechanism to iterate over mro?

File: lib/dl_core/dl_core_tests/unit/us_manager/test_migration.py [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        return entry
+
+    @staticmethod
+    def _migrate_v2_to_v3(entry: dict) -> dict:
+        entry["data"]["l1_field"] = "added_in_l1"
+        return entry
+
+
+@attr.s
+class Level2EntrySchemaMigration(Level1EntrySchemaMigration):
```
Comment (ovsds) [inline lib/dl_core/dl_core_tests/unit/us_manager/test_migration.py:39]: why is it child of l1?

File: lib/dl_core/dl_core/us_manager/schema_migration/base.py (hunk 1 above)
Comment (ForrestGump) [reply lib/dl_core/dl_core/us_manager/schema_migration/base.py:39]: imho, using `__mro__` attribute is much clearer vs `super()` in this case

File: lib/dl_core/dl_core/us_manager/schema_migration/base.py [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+                continue
+            if migration.version in seen_versions:
+                raise ValueError(f"Double migration detected for migration version: {migration.version}")
+            seen_versions.add(migration.version)
+            LOGGER.info(f"Apply migration ver={migration.version}, {migration.name}")
+            entry = migration.migrate(entry)
+        return entry
+
+    def migrate(self, entry: dict) -> dict:
+        original_entry = deepcopy(entry)
```
Comment (ovsds) [inline lib/dl_core/dl_core/us_manager/schema_migration/base.py:83]: we can move it inside except block, otherwise it's left unused

**Review by ovsds — APPROVED** [review]

**Pull request merged**